### ISSUE TITLE
Copy polyfills source maps

### ISSUE THREAD

**Issue by Joozty** [zeit/next.js#12861]
# Feature request

I would like to request a feature to copy source maps from polyfills to next build. Currently, the source maps are skipped but js file still contains a reference to it.

This request is very related to https://github.com/zeit/next.js/issues/7579 but I decided to report it separately since it probably requires a different bunch of work. Feel free to close if you find it duplicate.

## Is your feature request related to a problem? Please describe.
We use [source map loader](https://webpack.js.org/loaders/source-map-loader/) in our app which allows us to include third-party source maps in our build. It is extremely helpful especially in conjunction with [Sentry](https://sentry.io/welcome/). It helps us track down bugs easily and eventually if it is not a bug on our side then report it to correct package maintainers.

I prepared [MWE for you](https://github.com/Joozty/next-source-maps) when you run the app and open chrome console you will see something like:

```
Can't resolve './unfetch.js.map' in '/mnt/c/Users/joozty/Documents/next-source-maps/node_modules/next/dist/build/polyfills'
```

![sm](https://user-images.githubusercontent.com/12775440/81906212-34325b00-95c6-11ea-8ea3-daae27beb6aa.png)

it happens because unfetch.js file [copied here](https://github.com/zeit/next.js/blob/canary/packages/next/taskfile.js#L23) contains a reference to a source map which is not copied. (It's probably the same for other polyfills too.)

## Describe the solution you'd like

I would like to have these source maps included. It's not a big deal and this warning can be ignored but as I mentioned earlier it's a nice feature for having bug free JS ecosystem. :)

**Comment by timneutkens** [comment]
Feel free to investigate/fix this.

**Comment by maxigimenez** [comment]
@timneutkens I would love to take this one, one approach could be copying the `.map` file onto the build, what do you think?

**Comment by sophiekoonin** [comment]
FWIW, I think this might be because the build manifest contains a reference to the source map - another approach could be to make sure that when the build manifest is created, we're not including `*.map` files in the `_polyfills` section.

**Comment by maxigimenez** [comment]
> FWIW, I think this might be because the build manifest contains a reference to the source map - another approach could be to make sure that when the build manifest is created, we're not including `*.map` files in the `_polyfills` section.

Good point, as far as I saw next.js is just copying the files, not building `unfetch`, so in theory there's no control if the "sourceMappingURL" is on place or not

**Comment by Joozty** [comment]
Just for the sake of completeness, the new nextjs version brings more files with reference to non existing source-maps

![missing-source-maps](https://user-images.githubusercontent.com/12775440/89215866-ac0e9100-d5c9-11ea-907e-c08be889ddec.png)

**Comment by Joozty** [comment]
For anybody affected by this issue. You may disable source map preloader for these specific modules/files like this:

```
    config.module.rules.push(
      {
        test: /.*\.js$/,
        // disable these modules because they generate a lot of warnings about
        // non existing source maps
        // we cannot filter these warnings via config.stats.warningsFilter
        // because Next.js doesn't allow it
        // https://github.com/vercel/next.js/pull/7550#issuecomment-512861158
        // https://github.com/vercel/next.js/issues/12861
        exclude: [
          /@next\/react-dev-overlay/,
          /@next\/react-refresh-utils\/runtime.js$/,
          /@next\/react-refresh-utils\/internal\/helpers.js$/,
          /next\/dist\/build\/polyfills\/unfetch.js$/,
          /next\/dist\/build\/polyfills\/unfetch.js$/,
        ],
        use: ['source-map-loader'],
        enforce: 'pre',
      }
    )
```
